### PR TITLE
fix(metrics): Fix transaction metrics naming inconsistencies [INGEST-1511]

### PR DESCRIPTION
**Pull request by jjbayer**
Align the metric names (MRIs) in the project config allow list with the names that Relay actually tries to extract.

Do not merge before https://github.com/getsentry/sentry/pull/37045 is deployed.

### CODE REVIEW

File: src/sentry/relay/config/__init__.py [hunk 1]
```diff
@@ -410,7 +410,7 @@ def _filter_option_to_config_setting(flt, setting):
         "d:transactions/measurements.lcp@millisecond",
         "d:transactions/measurements.app_start_cold@millisecond",
         "d:transactions/measurements.app_start_warm@millisecond",
-        "d:transactions/measurements.cls@millisecond",
+        "d:transactions/measurements.cls@none",
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:413]: Unfortunately we cannot create a static string in the indexer for this, because `d:transactions/measurements.cls@none` was already added for multiple organizations as a "custom" measurement (we allow up to five custom measurements per transaction, and this was treated as such because of the different unit).

File: src/sentry/relay/config/__init__.py [hunk 2]
```diff
@@ -474,7 +474,7 @@ def get_transaction_metrics_settings(
                 assert breakdown_config["type"] == "spanOperations"
 
                 for op_name in breakdown_config["matches"]:
-                    metrics.append(f"d:transactions/breakdowns.ops.{op_name}")
+                    metrics.append(f"d:transactions/breakdowns.span_ops.ops.{op_name}")
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:477]: This is how Relay extracts it, see https://github.com/getsentry/relay/blob/bad9c5969800e35dd52e8b702373ab61b348ad2a/relay-server/src/metrics_extraction/transactions.rs#L578.

File: src/sentry/sentry_metrics/indexer/strings.py [hunk 3]
```diff
@@ -55,10 +55,10 @@
...
     "d:transactions/measurements.ttfb@millisecond": PREFIX + 118,
     "d:transactions/measurements.ttfb.requesttime@millisecond": PREFIX + 119,
-    "d:transactions/breakdowns.span_ops.http@millisecond": PREFIX + 120,
-    "d:transactions/breakdowns.span_ops.db@millisecond": PREFIX + 121,
-    "d:transactions/breakdowns.span_ops.browser@millisecond": PREFIX + 122,
-    "d:transactions/breakdowns.span_ops.resource@millisecond": PREFIX + 123,
+    "d:transactions/breakdowns.span_ops.ops.http@millisecond": PREFIX + 120,
+    "d:transactions/breakdowns.span_ops.ops.db@millisecond": PREFIX + 121,
+    "d:transactions/breakdowns.span_ops.ops.browser@millisecond": PREFIX + 122,
+    "d:transactions/breakdowns.span_ops.ops.resource@millisecond": PREFIX + 123,
```
Comment (jjbayer) [inline src/sentry/sentry_metrics/indexer/strings.py:61]: It should be okay to rewrite these static strings. These MRIs were never on the allow list, so Relay never extracted them. This change should go into a separate PR that is deployed before this one, though.

File: tests/relay_integration/test_metrics_extraction.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            }
+
+            consumer = kafka.Consumer(settings)
+            consumer.assign([kafka.TopicPartition("ingest-metrics", 0)])
+
+            self.post_and_retrieve_event(event_data)
+
+            metrics_emitted = set()
+            for _ in range(1000):
+                message = consumer.poll(timeout=0.1)
```
Comment (jjbayer) [inline tests/relay_integration/test_metrics_extraction.py:75]: Not sure if this is the best way to test what metrics relay emits. I saw a lot of duplicate messages, hence the `set()`.

File: src/sentry/relay/config/__init__.py [hunk 5]
```diff
@@ -474,7 +474,7 @@ def get_transaction_metrics_settings(
                 assert breakdown_config["type"] == "spanOperations"
 
                 for op_name in breakdown_config["matches"]:
-                    metrics.append(f"d:transactions/breakdowns.ops.{op_name}")
+                    metrics.append(f"d:transactions/breakdowns.span_ops.ops.{op_name}@millisecond")
```
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:477]: This is how Relay extracts it, see https://github.com/getsentry/relay/blob/bad9c5969800e35dd52e8b702373ab61b348ad2a/relay-server/src/metrics_extraction/transactions.rs#L578.

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (jjbayer) [inline src/sentry/relay/config/__init__.py:413]: CLS is unitless, see https://web.dev/cls/.

**Review by iker-barriocanal — APPROVED** [review]

File: src/sentry/search/events/constants.py [hunk 6]
```diff
@@ -209,10 +209,10 @@ class ThresholdDict(TypedDict):
...
     "measurements.ttfb": "d:transactions/measurements.ttfb@millisecond",
     "measurements.ttfb.requesttime": "d:transactions/measurements.ttfb.requesttime@millisecond",
-    "spans.browser": "d:transactions/breakdowns.span_ops.browser@millisecond",
-    "spans.db": "d:transactions/breakdowns.span_ops.db@millisecond",
-    "spans.http": "d:transactions/breakdowns.span_ops.http@millisecond",
-    "spans.resource": "d:transactions/breakdowns.span_ops.resource@millisecond",
+    "spans.browser": "d:transactions/breakdowns.span_ops.ops.browser@millisecond",
+    "spans.db": "d:transactions/breakdowns.span_ops.ops.db@millisecond",
+    "spans.http": "d:transactions/breakdowns.span_ops.ops.http@millisecond",
+    "spans.resource": "d:transactions/breakdowns.span_ops.ops.resource@millisecond",
```
[the remark this answers is not in the document]
Comment (jjbayer) [reply src/sentry/search/events/constants.py:215]: Sounds good, although I would not know how to enforce that at the moment.

File: src/sentry/relay/config/__init__.py (hunk 1 above)
Comment (andriisoldatenko) [reply src/sentry/relay/config/__init__.py:413]: https://github.com/getsentry/sentry/blob/be6c69a31f2343a870b5e58a5c0cf253f2bb52f4/src/sentry/snuba/metrics/naming_layer/mri.py#L34-L34 we have TODO here for better check `unit`.

File: src/sentry/search/events/constants.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply src/sentry/search/events/constants.py:215]: yeah i think it's a good idea too, we can adjust this regular expression https://github.com/getsentry/sentry/blob/be6c69a31f2343a870b5e58a5c0cf253f2bb52f4/src/sentry/snuba/metrics/naming_layer/mri.py#L34-L34.

**Review by andriisoldatenko — APPROVED** [review]

File: src/sentry/search/events/constants.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (jjbayer) [reply src/sentry/search/events/constants.py:215]: In principle, the `<name>` part of the MRI is free-form, so what would a regex to check the name look like? I think the best way to prevent wrong MRIs is to define them only once and then use that constant / enum everywhere. In fact, an enum like that already exists: https://github.com/getsentry/sentry/blob/3c5d227638be1456feac761364c8422c33a51122/src/sentry/snuba/metrics/naming_layer/mri.py#L70

[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply src/sentry/search/events/constants.py:215]: yeah but sometimes we want to convert from string into Enum, that's why we have this RegExp.

**Review by untitaker — APPROVED** [review]
